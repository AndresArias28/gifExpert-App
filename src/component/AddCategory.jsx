import { useState } from "react"

export const AddCategory = ({addCategories}) => {//Desestrucuturacion de las props para utilizar setCategories

    const [inputValue, setinputValue]  = useState('');

    //Esta funciÃ³n se encarga de actualizar el estado de inputValue segun lo que se escriba
    const onInputChange = ({target}) => {//Aqui se realiza la desestructuracion del objeto event y obtener la propiedad 'target'
        //console.log(event.target.value);
        setinputValue(target.value);//actualiza  el estado de inputValue con el valor actual del campo de entrada
    }

    //evento onsubmit maneja la presentacion del formulario, se ejecutara al enviar el formulario
    const onSubmit = (event) => {
        event.preventDefault(); //Evitar que se refresca el navegador cada vez que ingreso una letra
        if( inputValue.trim().length <= 1) return; //si solo hay 1 o ningun carater se sale de la funcion para que no acurra nada
        addCategories( inputValue.trim());//agrega categoria sin espacios
        setinputValue('');//refresco el navegador
    }

  return (

    <form onSubmit={(event) => onSubmit(event) }>
        <input  
            type="text" // indica caja de texto
            placeholder="buscar gif" //texto que aparece en el cuadro de texto
            value = { inputValue } /* en este caso el valor que ingresa el usuario se guardara en este estado/variable*/
            onChange={ onInputChange } //evento que se dispara cada vez que el valor de entrada cambia
        />
    </form>
  )
}
