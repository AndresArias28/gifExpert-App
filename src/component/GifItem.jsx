import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {//desestrucutra las props(que contiene todo los atributos esparcidos)
    
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
    )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
