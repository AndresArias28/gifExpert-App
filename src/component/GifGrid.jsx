import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ categoria }) => {//recibe categoria en las props del componente en forma desestructurada

    const {images, isLoading} = useFetchGifs( categoria );//crea un estado llamado images que se inicializa con catgegoria. Devuelve dos propiedades

    return(
        
        <>
            <h3> {categoria} </h3>
            {
                isLoading && (<h2>Cargando...</h2>  /*si  isLoading existe entonces mostrar cargando */) 
            }
            
            {/*si existe estado en image, se mapea el estado y se envian como las props para el componente GifItem. se le envia
             el key y las demas propiedades que contiene el estado image  */}
            <div className="card-grid">
                {
                    images?(
                        images.map( (image) => (
                            <GifItem key={ image.id } { ...image } />
                        )) 
                    ): (
                        <p>Cargando...</p>)
                }
            </div>
        </>                                                                                                                                                                                                                                                                                                                        
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired,
}