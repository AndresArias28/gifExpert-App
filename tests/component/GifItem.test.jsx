import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/component/GifItem";

describe("pruebas en GifItem", function() {

    const title = 'saitama';
    const url = 'http://one-punch.com/saitama.jpg';

    test('debe hacer match con el snapshoot', () =>{
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y yl ALT indicado', () =>{
        render(<GifItem title={title} url={url} />);
       // expect( screen.getByRole('img').src).toBe( url );
       const {src, alt} = screen.getByRole('img');
       expect(src).toBe( url );
       expect(alt).toBe( title)

    });

});