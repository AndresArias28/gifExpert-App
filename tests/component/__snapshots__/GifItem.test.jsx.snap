// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem debe hacer match con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
